### PR TITLE
Create opa-configmap.yaml

### PR DESCRIPTION
**Pull request by ch-stark**
in this example
- policies are in namespace policies
- opa-configmap gets generated in namespace opa (which is generated by the policy as well)

### CODE REVIEW

**Comment by ycao56** [comment]
@ch-stark please provide also add a link to this policy in https://github.com/open-cluster-management/policy-collection/blob/master/community/README.md it's also required to provide a description and link to instructions on how to use the policy

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: policy-opa
+  namespace: policies
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml]: remove hard coded namespace

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  annotations:
+    policy.open-cluster-management.io/standards: NIST-CSF
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml]: map to NIST SP 800-53

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+metadata:
+  name: placement-policy-opa
+  namespace: policies
+spec:
+  clusterConditions:
+  - status: "True"
+    type: ManagedClusterConditionAvailable
+  clusterSelector:
+    matchExpressions:
+      []  # selects all clusters if not specified
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml]: use environment:dev label selector https://github.com/open-cluster-management/policy-collection/blob/master/community/CM-Configuration-Management/policy-trusted-container.yaml#L54

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+subjects:
+- name: policy-opa
+  kind: Policy
+  apiGroup: policy.open-cluster-management.io
+---
+apiVersion: apps.open-cluster-management.io/v1
+kind: PlacementRule
+metadata:
+  name: placement-policy-opa
+  namespace: policies
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml]: remove hard coded namespace

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                kind: ConfigMap
+                metadata:
+                  name: nopod
+                  namespace: opa
+---
+apiVersion: policy.open-cluster-management.io/v1
+kind: PlacementBinding
+metadata:
+  name: binding-policy-opa
+  namespace: policies
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml]: remove hard coded namespace

**Comment by ch-stark** [comment]
implemented suggestions and updated readme

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,101 @@
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: policy-opa
+  annotations:
+    policy.open-cluster-management.io/standards: NIST SP 800-53
+    policy.open-cluster-management.io/categories: PR.IP Information Protection Processes and Procedures, PR.PT Protective Technology
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml]: these needs to be nist 800-53 cateogies and controls as well.

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    policy.open-cluster-management.io/controls: PR.IP-1 Baseline Configuration, PR.PT-3 Least Functionality
+spec:
+  remediationAction: enforce
+  disabled: false
+  policy-templates:
+    - objectDefinition:
+        apiVersion: policy.open-cluster-management.io/v1
+        kind: ConfigurationPolicy
+        metadata:
+          name: policy-opa-policies
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml:17]: probably name it more meaningful. e.g name it after what policy does

Comment (ycao56) [reply community/CM-Configuration-Management/opa-configmap.yaml:17]: seems this one is does the same thing as line 35

File: community/README.md [hunk 8]
```diff
@@ -10,6 +10,7 @@ View a list of policies that are organized by the security control catalog.
 * Security Assessment and Authorization
 * Configuration Management
   * [Trusted Container Policy](./CM-Configuration-Management/policy-trusted-container.yaml) -- Detect if running pods are using trusted images. For more information, please read [Trusted Container Policy Controller](https://github.com/ycao56/trusted-container-policy-controller).
+  * [OPA-Configmap Policy](./CM-Configuration-Management/opa-configmap.yaml). This policy has two purposes: Showing an Example how a ConfigMap can be created via Policies. It also ensures that the namespaces for OPA-installation and for the Policy-Placement already exist. Second it shows how to Put a `REGO` script into a Config-Map which can be evaluated by Open-Policy-Agent. For me information on this approach please check the [example repository](https://github.com/ycao56/mcm-opa).
```
Comment (ycao56) [inline community/README.md]: @dockerymick can you review the text here?

Comment (ycao56) [reply community/README.md]: @ch-stark I suggest also mention opa needs to be installed first on managed cluster as a prereq

**Review by dockerymick — APPROVED** [review]
Made a few changes on line 12 and 13, this looks good to me

File: community/CM-Configuration-Management/opa-configmap.yaml (hunk 6 above)
Comment (ch-stark) [reply community/CM-Configuration-Management/opa-configmap.yaml]: All policies have those values:  I'll change it to the same values
 policy.open-cluster-management.io/standards: NIST SP 800-53
 policy.open-cluster-management.io/categories: SI System and Information Integrity
 policy.open-cluster-management.io/controls: SI-5 Security Alerts, Advisories, and Directives

File: community/README.md [hunk 9]
```diff
@@ -9,7 +9,8 @@ View a list of policies that are organized by the security control catalog.
...
+  * [Trusted Container policy](./CM-Configuration-Management/policy-trusted-container.yaml): Use the trusted container policy to detect if running pods are using trusted images. For more information, see [Trusted Container Policy Controller](https://github.com/ycao56/trusted-container-policy-controller).
+  * [OPA ConfigMap policy](./CM-Configuration-Management/opa-configmap.yaml): Use the Open Policy Agent (OPA) ConfigMap policy to view example of how a ConfigMap can be created with a policy. This policy also verifies that the namespaces for your OPA installation and for the placement policy already exist. You can also view an example of adding a `REGO` script into a ConfigMap, which is evaluated by the OPA. For more information on this approach, see the [example repository](https://github.com/ycao56/mcm-opa). Also please note the OPA needs to be installed if you want to use this example in Context of OPA.
```
Comment (dockerymick) [inline community/README.md]: @ch-stark 
Can you update this statement: "Also please note the OPA needs to be installed if you want to use this example in Context of OPA."

to 

"**Note**: OPA must be installed to use the OPA ConfigMap policy"

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 10]
```diff
@@ -0,0 +1,101 @@
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: policy-opa
+  annotations:
+    policy.open-cluster-management.io/standards: NIST SP 800-53
+    policy.open-cluster-management.io/categories: SI System and Information Integrity
+    policy.open-cluster-management.io/controls: SI-5 Security Alerts, Advisories, and Directives
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml]: based on content of this policy, I think below is more properiate
```
    policy.open-cluster-management.io/categories: CM Configuration Management
    policy.open-cluster-management.io/controls: CM-2 Baseline Configuration
```

File: community/CM-Configuration-Management/opa-configmap.yaml [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+        spec:
+          remediationAction: enforce # will be overridden by remediationAction in parent policy
+          severity: high
+          namespaceSelector:
+            exclude: ["kube-*"]
+            include: ["default"]
+          object-templates:
+            - complianceType: musthave
+              objectDefinition:
+                kind: Namespace # must have namespace 'policies'
```
Comment (ycao56) [inline community/CM-Configuration-Management/opa-configmap.yaml]: I am still confused. Why do you need to create two namespaces? it's not used by this policy at all

Comment (ch-stark) [reply community/CM-Configuration-Management/opa-configmap.yaml]: OK I can delete them.  I wanted to ensure that the namespaces do exist

File: community/README.md [hunk 12]
```diff
@@ -9,7 +9,8 @@ View a list of policies that are organized by the security control catalog.
...
 * Configuration Management
-  * [Trusted Container Policy](./CM-Configuration-Management/policy-trusted-container.yaml) -- Detect if running pods are using trusted images. For more information, please read [Trusted Container Policy Controller](https://github.com/ycao56/trusted-container-policy-controller).
+  * [Trusted Container policy](./CM-Configuration-Management/policy-trusted-container.yaml): Use the trusted container policy to detect if running pods are using trusted images. For more information, see [Trusted Container Policy Controller](https://github.com/ycao56/trusted-container-policy-controller).
+  * [OPA ConfigMap policy](./CM-Configuration-Management/opa-configmap.yaml): Use the Open Policy Agent (OPA) ConfigMap policy to view example of how a ConfigMap can be created with a policy. This policy also verifies that the namespaces for your OPA installation and for the placement policy already exist. You can also view an example of adding a `REGO` script into a ConfigMap, which is evaluated by the OPA. For more information on this approach, see the [example repository](https://github.com/ycao56/mcm-opa). Note: OPA must be installed to use the OPA ConfigMap policy.
```
Comment (ycao56) [inline community/README.md]: calling it `OPA ConfigMap policy` is confusing. I think we should just call it `OPA Sample policy` and then explain what this sample policy actually does in the description. e.g. This policy defines an OPA policy using Rego scripts to block any pod in namespace `opa` that uses an image starts with `hooli.com` from being deployed

**Comment by ycao56** [comment]
@ch-stark this pr is not targeted to master branch and is out of date. can you update to latest and target it to master?

**Comment by ch-stark** [comment]
changed to master

**Review by ycao56 — APPROVED** [review]

**Pull request merged**